crates/alpen-ee/common/src/traits/prover.rs

```diff
@@ -0,0 +1,39 @@
+use async_trait::async_trait;
+
+use crate::{EeUpdateId, Proof, ProofId};
+
+#[derive(Debug)]
+pub enum ProofGenerationStatus {
+    /// Proof generation started or proof is getting generated
+    Pending,
+    /// Proof is ready and can be fetched using proof_id
+    Ready { proof_id: ProofId },
+    /// Proof generation has not been triggered for provided ee_update_id.
+    Invalid,
+    /// Cannot generate proof for some reason. All retries exhausted, etc.
+    Failed { reason: String },
+}
+
+/// Interface between Prover and Batch assembly
+/// Note: In case of reorgs, `EeUpdateId` is unique enough to not conflict between forks.
+/// Note2: eyre::Result<> is here mainly to pass back db related and other errors.
```

Comment:
This does not belong to the docstring.
Or at least you could say something like

"Most of the return types will have transparent errors from ..."